src/client/opamCommands.ml

```diff
@@ -394,22 +403,32 @@ let init cli =
     OpamStd.Exn.finally (fun () -> OpamRepositoryState.drop rt)
     @@ fun () ->
     if no_compiler then () else
-    let invariant, name =
+    let invariant, default_compiler, name =
       match compiler with
       | Some comp when String.length comp > 0 ->
-        OpamSwitchCommand.guess_compiler_invariant rt [comp], comp
-      | _ -> default_compiler, "default"
+        OpamSwitchCommand.guess_compiler_invariant rt [comp],
+        [],
+        comp
+      | _ ->
+        OpamFile.Config.default_invariant gt.config,
+        default_compiler, "default"
     in
-    OpamConsole.header_msg "Creating initial switch %s (%s)"
+    OpamConsole.header_msg "Creating initial switch '%s' (invariant %s%s)"
       name
       (match invariant with
        | OpamFormula.Empty -> "empty"
-       | c -> OpamFileTools.dep_formula_to_string c);
+       | c -> OpamFileTools.dep_formula_to_string c)
+      (match default_compiler with
+       | [] -> ""
+       | comp -> " - initially with "^ (OpamFormula.string_of_atoms comp));
     let (), st =
       try
         OpamSwitchCommand.create
           gt ~rt ~invariant ~update_config:true (OpamSwitch.of_string name) @@
-        (fun st -> (), OpamSwitchCommand.install_compiler st)
+        (fun st ->
+           (),
+           OpamSwitchCommand.install_compiler st
+             ~additional_installs:default_compiler)
```

Comment:
This causes an interactive prompt in `OpamSwitchCommand.install_compiler`

Reply:
Ah, well spotted, indeed that's unfortunate.